ament_cmake_test/cmake/run_test.py

```diff
@@ -129,20 +129,36 @@ def log(msg, **kwargs):
         env = dict(os.environ)
         if args.env:
             log('-- run_test.py: extra environment variables:')
+            previous_key = None
             for env_str in args.env:
-                key, value = separate_env_vars(env_str, 'env', parser)
+                # if CMake has split a single value containing semicolons
+                # into multiple arguments they are put back together here
+                if previous_key and '=' not in env_str:
+                    key = previous_key
+                    value = env[key] + ';' + env_str
+                else:
+                    key, value = separate_env_vars(env_str, 'env', parser)
                 log(' - {0}={1}'.format(key, value))
```

Comment:
This log output is going to be strange right? It would print for each "token" in a path for an env variable, where before it would have printed the combined path.